src/swap/components/SwapSettingsSlippageInput.tsx

```diff
@@ -0,0 +1,123 @@
+import { useCallback, useState } from 'react';
+import { background, border, cn, color, pressable } from '../../styles/theme';
+import type { SwapSettingsSlippageInputReact } from '../types';
+import { useSwapContext } from './SwapProvider';
+
+const slippageSettings = ['Auto', 'Custom'];
+
+export function SwapSettingsSlippageInput({
+  className,
+  defaultSlippage = 3,
+}: SwapSettingsSlippageInputReact) {
+  const { setLifeCycleStatus } = useSwapContext();
+  const [slippage, setSlippage] = useState(defaultSlippage);
+  const [isAutoSlippageSetting, setIsAutoSlippageSetting] = useState(true);
+
+  const updateSlippage = useCallback(
+    (newSlippage: number) => {
+      setSlippage(newSlippage);
+      setLifeCycleStatus({
+        statusName: 'slippageChange',
+        statusData: { maxSlippage: newSlippage },
+      });
+    },
+    [setLifeCycleStatus],
+  );
+
+  // Handles user input for custom slippage
+  // Parses the input and updates slippage if valid
+  const handleSlippageChange = useCallback(
+    (newSlippage: string) => {
+      const newSlippageNumber = Number.parseFloat(newSlippage);
+      if (!Number.isNaN(newSlippageNumber)) {
+        updateSlippage(newSlippageNumber);
+      }
+    },
+    [updateSlippage],
+  );
+
+  // Toggles between auto and custom slippage settings
+  // Resets to default slippage when auto is selected
+  const handleSlippageSettingChange = useCallback(
+    (auto: boolean) => {
+      setIsAutoSlippageSetting(auto);
+      if (auto) {
+        updateSlippage(defaultSlippage);
+      }
+    },
+    [updateSlippage, defaultSlippage],
+  );
+
+  return (
+    <section
+      className={cn(
+        background.default,
+        border.defaultActive,
+        'flex items-center gap-2',
+        className,
+      )}
+    >
+      <fieldset
+        className={cn(
+          background.default,
+          border.defaultActive,
+          'flex h-9 flex-1 rounded-xl border p-1',
+        )}
+      >
+        <legend className="sr-only">Slippage Setting</legend>
+        {slippageSettings.map((slippageSetting) => (
+          <button
+            key={slippageSetting}
+            type="button"
+            className={cn(
+              pressable.default,
+              color.foreground,
+              'flex-1 rounded-lg px-3 py-1 font-medium text-sm transition-colors',
+              isAutoSlippageSetting === (slippageSetting === 'Auto')
```

Comment:
The only thing that is bothering me is this. 

In general I also don't like this (slippageSetting === 'Auto') check

Reply:
The thing is, you are doing `slippageSetting === 'Auto'` in multiple places. And in some other you have `isAutoSlippageSetting`.

It's just confusing.

Next PR simplify this all thing.